### PR TITLE
Implement ListPoolProposerSlashings and ListPoolVoluntaryExits in the beacon API

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

What does this PR do? Why is it needed?

This PR implements beacon API's `ListPoolProposerSlashings` and `ListPoolVoluntaryExits` methods according to the spec: https://ethereum.github.io/eth2.0-APIs/#/Beacon

Which issues(s) does this PR fix?

Part of #7510

Other notes for review

Very similar to #8492

### CODE REVIEW

File: beacon-chain/rpc/beaconv1/pool.go [hunk 1]
```diff
@@ -55,7 +55,23 @@ func (bs *Server) SubmitAttesterSlashing(ctx context.Context, req *ethpb.Atteste
...
 func (bs *Server) ListPoolProposerSlashings(ctx context.Context, req *ptypes.Empty) (*ethpb.ProposerSlashingPoolResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "beaconv1.ListPoolProposerSlashings")
+	defer span.End()
+
+	headState, err := bs.ChainInfoFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+	sourceSlashings := bs.SlashingsPool.PendingProposerSlashings(ctx, headState, true)
```
Comment (terencechain) [inline beacon-chain/rpc/beaconv1/pool.go:65]:
```suggestion
	sourceSlashings := bs.SlashingsPool.PendingProposerSlashings(ctx, headState, true /* unlimited return slashings */)
```

File: beacon-chain/rpc/beaconv1/pool.go [hunk 2]
```diff
@@ -67,7 +83,24 @@ func (bs *Server) SubmitProposerSlashing(ctx context.Context, req *ethpb.Propose
...
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "beaconv1.ListPoolVoluntaryExits")
+	defer span.End()
+
+	headState, err := bs.ChainInfoFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+
+	sourceExits := bs.VoluntaryExitsPool.PendingExits(headState, headState.Slot(), true)
```
Comment (terencechain) [inline beacon-chain/rpc/beaconv1/pool.go:94]: Same. A comment after `true` will be nice

File: proto/migration/migration.go [hunk 3]
```diff
@@ -93,3 +93,45 @@ func V1Alpha1AttSlashingToV1(v1alpha1Slashing *ethpb_alpha.AttesterSlashing) *et
 		Attestation_2: V1Alpha1IndexedAttToV1(v1alpha1Slashing.Attestation_2),
 	}
 }
+
```
Comment (terencechain) [inline proto/migration/migration.go:96]: Some tests for these?

Comment (rkapka) [reply proto/migration/migration.go:96]: Yeah, the `migration` package doesn't have any tests... Would it be OK if I create a separate PR with tests for the whole package?

Comment (terencechain) [reply proto/migration/migration.go:96]: Sounds good 👍

**Review by terencechain — APPROVED** [review]

**Pull request merged**